plugins/guests/redhat/cap/rhn_register.rb

```diff
@@ -0,0 +1,121 @@
+module VagrantPlugins
+  module GuestRedHat
+    module Cap
+      class RhnRegister
+        # Test that the machine is already registered
+        def self.rhn_register_registered?(machine)
+          true if machine.communicate.execute('/usr/sbin/rhn_check', sudo: true)
+        rescue
+          false
+        end
+
+        # Test that we have rhn installed
+        def self.rhn_register(machine)
+          machine.communicate.test('/usr/sbin/rhn_check --version', sudo: true) &&
+            machine.communicate.test('/usr/sbin/rhnreg_ks --version', sudo: true)
+        end
+
+        # Register the machine using 'rhnreg_ks' command, config is (Open)Struct
+        def self.rhn_register_register(machine, ui)
+          rhn_register_upload_certificate(machine, ui) if machine.config.registration.ca_cert
+          rhn_register_server_url(machine, ui) if machine.config.registration.serverurl
+          command = "rhnreg_ks #{configuration_to_options(machine.config.registration)}"
+          machine.communicate.execute("cmd=$(#{command}); if [ \"$?\" != \"0\" ]; then echo $cmd | grep 'This system is already registered' || (echo $cmd 1>&2 && exit 1) ; fi", sudo: true)
+        end
+
+        # Unregister the machine using 'rhn_unregister.py' resource script
+        def self.rhn_register_unregister(machine)
+          tmp = '/tmp/rhn_unregister'
+          username = machine.config.registration.username
+          password = machine.config.registration.password
+
+          machine.communicate.tap do |comm|
+            # No api call can be made without username/password
```

Comment:
The `register` to the Spacewalk/Satellite(5)/RHN Classic can be done in a non-interactive mode using `username`+`password` or `org`+`activationkey` and is added to the command line using the `configuration_to_options` method (same as for `subscription_manager`).

The `unregister` is done in two steps:
  1. just remove the `/etc/sysconfig/rhn/systemid`, otherwise guest machine still "thinks" it is a part of RHN network
  2. delete the system from the Spacewalk/Satellite(5)/RHN Classic. This can only be achieved using one of the options:
    - using the WebUI after a successful login with a valid username/password
    - with an API call to "system.deleteSystems" using a valid session key obtained with the call to `auth.login` that also requires a valid username/password 
  
  Unfortunately, both option to delete the system from Spacewalk/Satellite(5)/RHN Classic work with username/password. The guest machine will not be deleted from Spacewalk/Satellite(5)/RHN Classic by the `rhn_manager` manager if the `vagrant-register` plugin is configured only with org/activationkey.